src/Tester/DependencyResolverGrainTests.cs

```diff
@@ -0,0 +1,130 @@
+OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+using System.IO;
+using System.Threading.Tasks;
+using Autofac;
+using Autofac.Builder;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Orleans.Autofac;
+using Orleans.Providers;
+using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+using Orleans.Runtime.MembershipService;
+using Orleans.Runtime.ReminderService;
+using Orleans.Streams;
+using Orleans.TestingHost;
+using UnitTests.GrainInterfaces;
+using UnitTests.Grains;
+using UnitTests.Tester;
+
+namespace UnitTests.General
+{
+    /// <summary>
+    /// Summary description for SimpleGrain
+    /// </summary>
+    [DeploymentItem("OrleansDependencyResolverConfigurationForTesting.xml")]
+    [DeploymentItem("Orleans.Autofac.dll")]
+    [TestClass]
+    public class DependencyResolverGrainTests : UnitTestSiloHost
+    {
+        private const string SimpleGrainNamePrefix = "UnitTests.Grains.SimpleG";
+
+        public DependencyResolverGrainTests()
+            : base(new TestingSiloOptions
+            {
+                StartPrimary = true,
+                StartSecondary = false,
+                SiloConfigFile = new FileInfo("OrleansDependencyResolverConfigurationForTesting.xml")
+            })
+        {
+        }
+
+        public ISimpleGrain GetSimpleGrain()
+        {
+            return GrainFactory.GetGrain<ISimpleGrain>(GetRandomGrainId(), SimpleGrainNamePrefix);
+        }
+
+        private static int GetRandomGrainId()
+        {
+            return random.Next();
+        }
+
+        [ClassCleanup]
+        public static void MyClassCleanup()
+        {
+            StopAllSilos();
+        }
+
+        [TestMethod, TestCategory("BVT"), TestCategory("Functional")]
+        public async Task DISimpleGrainGetGrain()
+        {
+            ISimpleGrain grain = GetSimpleGrain();
+            int ignored = await grain.GetAxB();
+        }
+    }
+
+    public class AutofacDepdendencyResolverProvider : IDependencyResolverProvider
+    {
+        public string Name
+        {
+            get { return "Autofac DI"; }
+        }
+
+        public Task Init(string name, IProviderRuntime providerRuntime, IProviderConfiguration config)
+        {
+            //
+            // This method will not be called by the DependenceResolverProviderManager, when loading the provider.
+            //
+
+            throw new System.NotSupportedException();
+        }
+
+        public IDependencyResolver GetDependencyResolver(ClusterConfiguration config, NodeConfiguration nodeConfig, TraceLogger logger)
+        {
+            var builder = new ContainerBuilder();
+
+            builder.RegisterType(typeof(GrainBasedMembershipTable));
```

Comment:
is it possible to say: for all types that I did not explicitly specify via `builder.RegisterType` use the default constructor? That way we don't have to specify ALL grains and especially not system grains like those ones.